File: vul-mongo.java
package services;

import com.mongodb.MongoClient;
import com.mongodb.MongoClientURI;

public class A {
    protected DB() {
        // ruleid: java-mongo-hardcoded-secret
        new MongoClient(new MongoClientURI("mongodb://user:password@a.mongolab.com:37601/goparty"))
        // ok: java-mongo-hardcoded-secret
        new MongoClient(new MongoClientURI("mongodb://<user>:password@a.mongolab.com:37601/goparty"))     
        // ok: java-mongo-hardcoded-secret
        new MongoClient(new MongoClientURI("mongodb://<user>:<password>@a.a.com:37601/goparty"))    
        // ok: java-mongo-hardcoded-secret
        new MongoClient(new MongoClientURI("mongodb://a.mongolab.com:37601/goparty"))      

    }
}

File: vul-mongo.py
from pymongo import MongoClient

# ruleid: python-mongo-hardcoded-secret
MongoClient("mongodb://pretty:shop1234@ds139942.mlab.com:39942/moviebot")
# ok: python-mongo-hardcoded-secret
MongoClient("mongodb://a:<password>@.mlab.com:39942/moviebot")
# ok: python-mongo-hardcoded-secret
MongoClient("mongodb://<user>:a@aaaa.mlab.com:39942/moviebot")
# ok: python-mongo-hardcoded-secret
MongoClient("mongodb://user@aaaa.mlab.com:39942/moviebot")
